boot/makebootable.go

```diff
@@ -341,7 +357,11 @@ func makeRunnableSystem(model *asserts.Model, bootWith *BootableSet, sealer *Tru
 		currentTrustedBootAssets = sealer.currentTrustedBootAssetsMap()
 		currentTrustedRecoveryBootAssets = sealer.currentTrustedRecoveryBootAssetsMap()
 	}
-	recoverySystemLabel := filepath.Base(bootWith.RecoverySystemDir)
+	// filepath.Base("") returns ".", so we need to be a bit careful here
+	recoverySystemLabel := ""
+	if bootWith.RecoverySystemDir != "" {
+		recoverySystemLabel = filepath.Base(bootWith.RecoverySystemDir)
+	}
```

Comment:
do we actually need this change?

Reply:
that's fine, we probably want a small unit test for this case then

Reply:
Added now

Reply:
Nice catch! I think one of my earlier experiments had this problem too